### PR TITLE
Fix `no-invalid-double-slash-comments` reported ranges

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

See https://github.com/stylelint/stylelint/issues/7904

> Is there anything in the PR that needs further explanation?

I also noticed that rules were eagerly serialized, this is quite expensive.

This change:

```
;node benchmark-rule.mjs no-invalid-double-slash-comments 'true'                               
Warnings: 0
Mean: 95.35613458823532 ms
Deviation: 9.862092026552252 ms
```

Main:

```
;node benchmark-rule.mjs no-invalid-double-slash-comments 'true'
Warnings: 0
Mean: 118.80464825 ms
Deviation: 23.471924727353816 ms
```

### CODE REVIEW

File: lib/rules/no-invalid-double-slash-comments/__tests__/index.mjs [hunk 1]
```diff
@@ -75,6 +89,33 @@ testRule({
...
+					color: red;
+				}
+			`,
+			description: 'regular comments interleaved with double slash comments',
+			warnings: [
+				{
+					message: messages.rejected,
+					line: 1,
+					column: 5,
+					endLine: 2,
```
Comment (romainmenke) [inline lib/rules/no-invalid-double-slash-comments/__tests__/index.mjs:107]: I am not sure if the `endLine` here is correct or helpful, but out of scope for this issue. This issue is limited to incorrect ranges as a result of interleaved comments. It might be that this is fine :)

File: lib/rules/no-invalid-double-slash-comments/index.cjs [hunk 2]
```diff
@@ -37,9 +37,9 @@ const rule = (primary) => {
 		});
 
 		root.walkRules((ruleNode) => {
-			// ruleNode.selectors elements are trimmed
```
Comment (romainmenke) [inline lib/rules/no-invalid-double-slash-comments/index.cjs:40]: I removed this comment because there is test coverage for this aspect and we already have guidance against using `node.selectors`.

File: lib/rules/no-invalid-double-slash-comments/index.mjs [hunk 3]
```diff
@@ -33,9 +33,9 @@ const rule = (primary) => {
 		});
 
 		root.walkRules((ruleNode) => {
-			// ruleNode.selectors elements are trimmed
-			const selectors = ruleNode.selector.split(',');
-			const ruleStringified = ruleNode.toString();
+			const selectors = (ruleNode.raws?.selector?.raw ?? ruleNode.selector).split(',');
```
Comment (ybiquitous) [inline lib/rules/no-invalid-double-slash-comments/index.mjs:36]: getRuleSelector util seems available instead.

File: lib/rules/no-invalid-double-slash-comments/index.cjs (hunk 2 above)
Comment (Mouvedia) [reply lib/rules/no-invalid-double-slash-comments/index.cjs:40]: > we already have guidance against using `node.selectors`

I didn't know about that. If we already have a regression test coverage it can be removed.
i.e. the comment was there so that no one would refactor it back

**Review by ybiquitous — APPROVED** [review]
Thank you for your quick fix! I left a review but LGTM 👍

File: lib/rules/no-invalid-double-slash-comments/index.mjs (hunk 3 above)
Comment (romainmenke) [reply lib/rules/no-invalid-double-slash-comments/index.mjs]: Thank you @ybiquitous 🙇

Maybe interesting to have some custom eslint rules to check such stuff :)

File: lib/rules/no-invalid-double-slash-comments/__tests__/index.mjs (hunk 1 above)
Comment (Mouvedia) [reply lib/rules/no-invalid-double-slash-comments/__tests__/index.mjs:107]: It has something to do with the trick of using `{}`.
i.e. `// bar {}` vs `// bar`

Comment (Mouvedia) [reply lib/rules/no-invalid-double-slash-comments/__tests__/index.mjs:107]: It has something to do with the trick of using `{}`.
The others tests—except l74–83—all use a `{}` construct.
i.e. `// bar {}` vs `// bar`

So we should either have the real commented code (including the following construct) or the part before the `\n` selected.
i.e. either `// Comment 2\n` or `// Comment 2\n.c {\ncolor: red;\n}` (not `// Comment 2\n.`)
Either way it should be fixed.
From an UX standpoint it should be `// Comment 2\n`.

tl;dr `(lines.length > 1)` is false when using `{}`
